File: exercism/go/bob/bob.go
package bob

import (
	"regexp"
	"strings"
)

var yell regexp.Regexp = *regexp.MustCompile(`^[^a-z]*[A-Z]+[^a-z]*$`)

func Hey(remark string) string {
	remark = strings.TrimSpace(remark)
	isQuestion := strings.HasSuffix(remark, "?")
	isYelling := yell.MatchString(remark)
	switch {
	case isQuestion && isYelling:
		return "Calm down, I know what I'm doing!"
	case isYelling:
		return "Whoa, chill out!"
	case isQuestion:
		return "Sure."
	case remark == "":
		return "Fine. Be that way!"
	default:
		return "Whatever."
	}
}

File: exercism/zig/bob/bob.zig
const std = @import("std");

pub fn response(s: []const u8) []const u8 {
    const phrase = std.mem.trim(u8, s, &std.ascii.whitespace);
    const silence = phrase.len == 0;
    if (silence) return "Fine. Be that way!";

    const yell = is_uppercase(phrase);
    const question = phrase[phrase.len - 1] == '?';
    if (yell and question) return "Calm down, I know what I'm doing!";
    if (yell) return "Whoa, chill out!";
    if (question) return "Sure.";

    return "Whatever.";
}

fn is_uppercase(s: []const u8) bool {
    var res = false;
    for (s) |c| {
        if (std.ascii.isLower(c)) return false;
        if (std.ascii.isUpper(c)) res = true;
    }
    return res;
}
